### DISCUSSION TITLE
How to connect IR remote control ZS-06 to Zigbee2MQTT

### DISCUSSION THREAD

**Discussion by MartienV** [Koenkk/zigbee2mqtt#13516] [Devices]
Hallo Koen,

I bought at Aliexpress a ZS-06 IR remote control:
https://nl.aliexpress.com/item/1005003878194474.html?spm=a2g0o.order_list.0.0.5b8179d2ylrbQB&gatewayAdapt=glo2nld

In my zigbee2mqtt log I see the remote control is paired.
In Domoticz, I see the device with an ID.

In this page I see some commands to switch the remote control to learning mode:  https://www.zigbee2mqtt.io/devices/ZS06.html

But now I am lost. I miss the following steps.
Where and how exactly do I give these commands?
Thank you so much for your help!

**Comment by LoKoNaoX** [comment]
did you find any tutorial, I am lost too. And I found this post, if you found a guide or tutorial, and you give me the link. Thanks you.

**Reply by MartienV** [reply]
In the zigbee2MQTT  log  I could see that the ZS-06 was paired.
With this command you can set the learn-mode to ON:
mosquitto_pub -t 'zigbee2mqtt/id_of_your_ZS-06/set' -m '{ "learn_ir_code": "ON" }'

Then you give the IR signal with the remote control of the device you want to capture.
In zigbee2MQTT log you can find the code which is captured.
Finally you can send the IR-code with your ZS-06 with this command:
mosquitto_pub -t 'zigbee2mqtt/0x94deb8fffe37cb22/set' -m '{"ir_code_to_send":"HERE_YOUR_IR_CODE_TO_SEND","linkquality":61}'

Sadly I couldn't succeed, because I want to use it with a lamp which sends different IR-codes.
So, every time I capture the IR-code for to  switch on/off   the IR-code is different.

**Reply by bvhoesel** [reply]
Thanks for the info. Nice 'deduction' for the mqtt command. Using it this way from Domoticz with custom shell script. As a hint, the linkquality parameter can be omitted. If used at the moment it gives an error in zibee2mqtt.
